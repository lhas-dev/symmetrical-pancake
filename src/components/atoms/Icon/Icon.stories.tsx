import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Icon } from "./Icon";

export default {
  title: "Storybook/Atoms/Icon",
  component: Icon,
  argTypes: {
    src: { control: "text" },
  },
} as ComponentMeta<typeof Icon>;

const Template: ComponentStory<typeof Icon> = (args) => <Icon {...args} />;

export const Default = Template.bind({});
Default.args = {
  src: "DashboardIcon",
};
