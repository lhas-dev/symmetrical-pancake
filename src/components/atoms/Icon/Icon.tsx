import React, { Suspense, useEffect, useState } from "react";

interface IconProps {
  src: string;
}

export const Icon = ({ src }: IconProps) => {
  const [path, setPath] = useState("");

  useEffect(() => {
    const dynamicPath = () => import(`../../../icons/${src}.svg`);
    const onReady = async () => {
      try {
        const response = await dynamicPath();
  
        setPath(response.default);
      } catch (e) {
        setPath("");
      }
    };
    
    onReady();
  }, [src]);

  return (
    <Suspense fallback={<div />}>
      <img src={path} className={src} alt="Icon" />
    </Suspense>
  );
};
