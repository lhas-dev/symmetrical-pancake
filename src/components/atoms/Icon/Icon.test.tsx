import { render, act } from "@testing-library/react";
import { Icon } from "./Icon";

test("renders Icon with valid params", async () => {
  const tree = await act(async () => {
    render(<Icon src="DashboardIcon" />);
  });
  expect(tree).toMatchSnapshot();
});

test("renders Icon with invalid params", async () => {
  const tree = await act(async () => {
    render(<Icon src="NotExistsIcon" />);
  });
  expect(tree).toMatchSnapshot();
});
