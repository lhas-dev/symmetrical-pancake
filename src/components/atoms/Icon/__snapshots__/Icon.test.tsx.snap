// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Icon with invalid params 1`] = `undefined`;

exports[`renders Icon with valid params 1`] = `undefined`;
